### PR TITLE
reselect for flow v0.54

### PR DESCRIPTION
**Pull request by julienw**
This fixes the error found in https://github.com/facebook/flow/issues/4959. This is _likely_ a Flow bug that can be fixed by a libdef change. Or this is a libdef bug that's got exposed by Flow's higher strictness.

### CODE REVIEW

File: definitions/npm/reselect_v3.x.x/flow_v0.34.x-/reselect_v3.x.x.js [hunk 1]
```diff
@@ -1,85 +1,144 @@
-declare module 'reselect' {
-  declare type Selector<TState, TProps, TResult> = {
-      (state: TState, props: TProps, ...rest: any[]): TResult;
+declare module "reselect" {
+  declare type Selector<-TState, TProps, TResult> = {
```
Comment (julienw) [inline definitions/npm/reselect_v3.x.x/flow_v0.34.x-/reselect_v3.x.x.js]: This is the only meaningful change. This comes from @calebmer in https://twitter.com/calebmer/status/915669046191824896

All other changes are prettier's work :)

File: definitions/npm/reselect_v3.x.x/test_reselect_3.x.x.js [hunk 2]
```diff
@@ -42,50 +39,69 @@ createSelector(
...
+// TEST: Should work when using another selector as functions
+const simpleSelector = (state: State): string => "foo";
+const resultSelector = (arg: string): string => "foo";
+
+const combinedSelector1 = createSelector(simpleSelector, resultSelector);
+
+const combinedSelector2: (state: State) => string = createSelector(
+  combinedSelector1,
+  resultSelector
+);
```
Comment (julienw) [inline definitions/npm/reselect_v3.x.x/test_reselect_3.x.x.js]: this is the only new test. This comes from https://github.com/facebook/flow/issues/4959

**Comment by julienw** [comment]
Maybe I should run prettier first, and apply my changes on top of it ?

**Comment by GAntoine** [comment]
Does #1316 affect this in any way? I'd like to merge it first (since it's older). Would you mind heading over and giving it a review?

**Comment by julienw** [comment]
I think it's one or the other but I didn't have the time to look at the
other yet. This will be for tomorrow !

Le 9 oct. 2017 21:40, "Georges-Antoine Assi" <notifications@github.com> a
écrit :

**Comment by julienw** [comment]
I looked at the other one, it still has issues for now.

Now to be completely honest I don't know why this PR fixes the problem. Maybe it's a wrong fix for a real bug. I'd like to ask @calebmer again before we merge it.

**Comment by GAntoine** [comment]
Caleb has his hands full this week with flow work, so I'm going to take your word (and his tweet) that the issue is fixed. If it comes up as an incorrect/broken fix, we'll address it then. Cheers!

**Pull request merged**